Upgrade HTPostingsFormat class using the corresponding PostingsFormat…

liseli opened the pull request:
This PR updates HTPostingFormat class to use the corresponding version of Lucene 8.11. The previous code was created used Lucene 5, then the code have to be upgrade to generate new jars compatibles with Solr 8.11. The README file was also updated

liseli committed:
Upgrade HTPostingsFormat class using the corresponding PostingsFormat package for lucene 8.11

mwarin reviewed (approved):
Clear & concise. Approved.

moseshll reviewed (approved):
I think would be worth considering a few further changes to the README and Java file, since this doc doesn't get updated very often:
1. For Java refugees like me (I deal with Java about once every 15 years) an explanation of how to do step 1 ("compile HTPostingsFormatWrapper.java"), i.e. for me it was download Lucene 8.11 and run `javac -cp lucene-8.11.2/core/lucene-core-8.11.2.jar HTPostingsFormatWrapper.java` but that was by trial and error.
2. It wouldn't hurt to word-wrap the existing README to 80 characters, for those of us who occasionally use `less` or `pico` on the README.
3. There are a couple typos in the added text (e.g. "jas")
4. That link to the original discussion that spawned this custom class (`http://lucene.472066.n3.nabble.com/How-to-configure-Solr-PostingsFormat-block-size-tt4179029.html` inside the .java file is now a dead link, apparently. I couldn't find anything on the Wayback Machine, so maybe the best thing to do is mark it up as "(dead link)"? I didn't check any of the other links in the README, so maybe some of them are outdated too. It's definitely not a big thing.

aelkiss commented on README.md:
```diff
@@ -36,16 +36,19 @@ See *Background details* (below) for more details
 
 ### Creating a special jar file.
 
+Before creating the jar file you should check the Java version accepted by Solr. For example, for Solr 8.11 the recommended version of Java is hihger that 9, however higher than 11 the jar failed, then the maximun version of java you can use for generating the jas is 11.
```
Perhaps we could just say:

For best results, compile using the same version of Java that you will use to run Solr. For Solr 8.11, JDK 11 is recommended.

aelkiss reviewed (approved):
Seems straightforward enough; just one suggestion about what we say about the Java version.

liseli committed:
Improved the documentation

liseli commented:
> I think would be worth considering a few further changes to the README and Java file, since this doc doesn't get updated very often:
> 
> 1. For Java refugees like me (I deal with Java about once every 15 years) an explanation of how to do step 1 ("compile HTPostingsFormatWrapper.java"), i.e. for me it was download Lucene 8.11 and run `javac -cp lucene-8.11.2/core/lucene-core-8.11.2.jar HTPostingsFormatWrapper.java` but that was by trial and error.
> 2. It wouldn't hurt to word-wrap the existing README to 80 characters, for those of us who occasionally use `less` or `pico` on the README.
> 3. There are a couple typos in the added text (e.g. "jas")
> 4. That link to the original discussion that spawned this custom class (`http://lucene.472066.n3.nabble.com/How-to-configure-Solr-PostingsFormat-block-size-tt4179029.html` inside the .java file is now a dead link, apparently. I couldn't find anything on the Wayback Machine, so maybe the best thing to do is mark it up as "(dead link)"? I didn't check any of the other links in the README, so maybe some of them are outdated too. It's definitely not a big thing.

Thanks for your comments Moses. I've reviewed the README file, fixed, typos and links. I have also added the command to generate the .class file. Some of the links seem does not exist anymore.
